src/main/java/no/nav/veilarboppfolging/repository/KvpRepository.java

```diff
@@ -132,13 +131,19 @@ public void stopKvp(long kvpId, AktorId aktorId, String avsluttetAv, String avsl
      * or zero if there is no current KVP period.
      */
     public long gjeldendeKvp(AktorId aktorId) {
-        try {
-            return db.queryForObject("SELECT gjeldende_kvp FROM oppfolgingstatus WHERE aktor_id = ?",
-                    (rs, row) -> rs.getLong("gjeldende_kvp"),
-                    aktorId.get());
-        } catch (EmptyResultDataAccessException e) {
-            return 0;
-        }
+        return queryForNullableObject(
+                () -> db.queryForObject("SELECT gjeldende_kvp FROM oppfolgingstatus WHERE aktor_id = ?",
+                (rs, row) -> rs.getLong("gjeldende_kvp"),
+                aktorId.get())
+        ).orElse(0L);
+    }
+
+    public Optional<KvpPeriodeEntity> hentGjeldendeKvpPeriode(AktorId aktorId) {
+        return queryForNullableObject(
+                () -> db.queryForObject("SELECT * FROM KVP WHERE aktor_id = ? AND avsluttet_dato IS NULL ORDER BY opprettet_dato FETCH NEXT 1 ROWS ONLY",
```

Comment:
Skulle ikke denne har 'ORDER BY opprettet_dato DESC', hvis du skal hente nyeste?

Reply:
Good catch. I teorien så skal ikke dette skje siden det burde kun finnes 1 kvp periode om gangen, men det er mye dårlig data ute å går så vi må likevel ta høyde for det. Skal rette på det 👍

Reply:
Eller er det kanskje riktigere å returnere den første gjeldende likevel? Slik at brukere med flere kvp-perioder ikke vil få en ny gjeldende kvp-periode hvis det skulle blitt opprettet en ny med et uhell